srv/salt/_modules/packagemanager.py

```diff
@@ -0,0 +1,274 @@
+from subprocess import Popen, PIPE
+from platform import linux_distribution
+import logging
+import os
+
+log = logging.getLogger(__name__)
+
+
+class PackageManager(object):
+
+    """
+    That is not the nativ salt module and is not ment to
+    replace it.
+    This module was created to react on packagemanagers
+    reboot advice.
+    """
+
+    def __init__(self, **kwargs):
+        self.debug = kwargs.get('debug', False)
+        self.kernel = kwargs.get('kernel', False)
+        self.reboot = kwargs.get('reboot', True)
+
+        self.platform = linux_distribution()[0].lower()
+        if "suse" in self.platform or "opensuse" in self.platform:
+            log.info("Found {}. Using {}".format(self.platform, Zypper.__name__))
+            self.pm = Zypper(**kwargs)
+        elif "fedora" in self.platform or "centos" in self.platform:
+            log.info("Found {}. Using {}".format(self.platform, Apt.__name__))
+            self.pm = Apt(**kwargs)
+        else:
+            raise ValueError("Failed to detect PackageManager for OS."
+                             "Open an issue on github.com/SUSE/DeepSea")
+
+    def reboot_in(self):
```

Comment:
I initially planned to make the reboot delay-time flexible as in:

```python
def reboot_in(self, time=1):
    cmd = "shutdown -r +{}".format(time)
```

but I don't see the use-case any longer.  -> changing back to _reboot(self)